### PR TITLE
Refactor set filter

### PR DESCRIPTION
**Pull request by JustinD85**

### CODE REVIEW

File: src/Components/App.js [hunk 1]
```diff
@@ -66,42 +67,32 @@ class App extends Component {
...
-      genre: {
-        strategy: false,
-        family: false,
-        party: false,
-        adventure: false
+    let { filterOptions } = this.state;
+
+    for (let filterOptionKey in filterOptions) {
+      for (let eachValuesKey in filterOptions[filterOptionKey]) {
+        filterOptions[filterOptionKey][eachValuesKey] = false;
```
Comment (foxwellm) [inline src/Components/App.js:74]: GREAT find here! Such an elegant way of resetting these values

File: src/Components/App.js [hunk 2]
```diff
@@ -178,61 +205,19 @@ class App extends Component {
...
-          || filterOptions.genre.adventure && game.genre_ID.includes(8)) {
-          return game
-        } else if (filterOptions.genre.strategy || filterOptions.genre.family || filterOptions.genre.party || filterOptions.genre.adventure) {
-          return
-        } else {
-          return game
-        }
+        return filterOptions.type.board ? game.genre_ID.includes(6)
+          : filterOptions.type.card ? game.genre_ID.includes(7)
+            : game
```
Comment (foxwellm) [inline src/Components/App.js:212]: I like this syntax for the nested ternary better, thanks

File: src/Components/App.js [hunk 3]
```diff
@@ -178,61 +205,19 @@ class App extends Component {
...
       })
+      .filter(game => this.filterPlayerCheck(filterOptions, game))
+      .filter(game => this.filterGameAgeCheck(filterOptions, game))
+      .filter(game => this.filterGenreCheck(filterOptions, game))
 
     if (inputValue) {
-      filteredGames = filteredGames.filter(game => game.game.toLowerCase().includes(inputValue))
+      filteredGames = filteredGames.filter(game => game.game
+        .toLowerCase().includes(inputValue))
         .sort((a, b) => a.game.localeCompare(b.game));
```
Comment (foxwellm) [inline src/Components/App.js:221]: I like the dropdown of .toLowerCase here to match the next stringed iterator

**Review by foxwellm — APPROVED** [review]

File: src/Components/App.js (hunk 1 above)
Comment (whitneyfirth) [reply src/Components/App.js:74]: Agreed, this is much more concise & readable in the long run!

File: src/Components/App.js (hunk 3 above)
Comment (whitneyfirth) [reply src/Components/App.js:221]: Agreed, nice thinking Justin. This entire function is much more readable and to the point.

**Review by whitneyfirth — APPROVED** [review]

**Pull request merged**